src/langchain_google_spanner/vector_store.py

```diff
@@ -0,0 +1,1181 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import json
+import logging
+import warnings
+import uuid
+import datetime
+import re
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Callable,
+    Iterable,
+    List,
+    Dict,
+    Optional,
+    Tuple,
+    Type,
+)
+from typing import Union
+from google.cloud.spanner_v1 import param_types
+from google.cloud.spanner_v1 import JsonObject
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from google.cloud import spanner
+from google.cloud.spanner import Client
+from google.cloud.spanner_v1.streamed import StreamedResultSet
+from langchain_core.utils import get_from_dict_or_env
+from langchain_core.vectorstores import VectorStore
+from enum import Enum
+import numpy as np
+from abc import ABC, abstractmethod
+from google.cloud.spanner_admin_database_v1.types import DatabaseDialect
+from langchain_community.vectorstores.utils import maximal_marginal_relevance
+from google.cloud.spanner_v1 import Type
+
+logger = logging.getLogger(__name__)
+
```

Comment:
Since we are waiting on a Spanner review, can we also add the user agent in this PR or separately:
```suggestion
from .version import __version__

USER_AGENT = "langchain-google-spanner-python:vectorstore" + __version__
```

Then update the logic for the client to append the user agent
https://github.com/googleapis/langchain-google-datastore-python/blob/3a62380794ad567ff8e9c0d5d7054093fb54cdb4/src/langchain_google_datastore/document_loader.py#L160-L168